### ISSUE TITLE
make clean misses files

### ISSUE THREAD

**Issue by HW42** [linux-test-project/ltp#559]
`make clean` currently doesn't clean up properly:

```
$ make autotools
[...]
$ ./configure 
[...]
$ make
[...]
$ make clean
[...]
$ git clean -dxn
Would remove aclocal.m4
Would remove autom4te.cache/
Would remove compile
Would remove confYSmH6M/
Would remove config.guess
Would remove config.log
Would remove config.status
Would remove config.sub
Would remove configure
Would remove execltp
Would remove gitignore_clean/
Would remove include/config.h
Would remove include/config.h.in
Would remove include/lapi/syscalls.h
Would remove include/mk/config-openposix.mk
Would remove include/mk/config.mk
Would remove include/mk/features.mk
Would remove include/stamp-h1
Would remove install-sh
Would remove lib/ltp.pc
Would remove m4/Makefile
Would remove m4/Makefile.in
Would remove m4/ltp-version.m4
Would remove missing
Would remove testcases/realtime/aclocal.m4
Would remove testcases/realtime/autom4te.cache/
Would remove testcases/realtime/compile
Would remove testcases/realtime/config.guess
Would remove testcases/realtime/config.sub
Would remove testcases/realtime/configure
Would remove testcases/realtime/include/realtime_config.h.in
Would remove testcases/realtime/install-sh
Would remove testcases/realtime/m4/Makefile.in
Would remove testcases/realtime/missing
Would remove utils/ffsb-6.0-rc2/.deps/
Would remove utils/ffsb-6.0-rc2/Makefile
Would remove utils/ffsb-6.0-rc2/autom4te.cache/
Would remove utils/ffsb-6.0-rc2/config.h
Would remove utils/ffsb-6.0-rc2/config.h.in~
Would remove utils/ffsb-6.0-rc2/config.log
Would remove utils/ffsb-6.0-rc2/config.status
Would remove utils/ffsb-6.0-rc2/stamp-h
Would remove utils/ffsb-6.0-rc2/stamp-h1
```

Depending on the timestamps (git doesn't preserve timestamps) sometimes `utils/ffsb-6.0-rc2/configure` gets regenerated. So even committed files are changed:

```
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   utils/ffsb-6.0-rc2/config.h.in
	modified:   utils/ffsb-6.0-rc2/configure

no changes added to commit (use "git add" and/or "git commit -a")
$ 
```

**Comment by pevik** [comment]
> Depending on the timestamps (git doesn't preserve timestamps) sometimes `utils/ffsb-6.0-rc2/configure` gets regenerated. So even committed files are changed:
> 
> ```
> $ git status
> On branch master
> Your branch is up-to-date with 'origin/master'.
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git checkout -- <file>..." to discard changes in working directory)
> 
> 	modified:   utils/ffsb-6.0-rc2/config.h.in
> 	modified:   utils/ffsb-6.0-rc2/configure

`ffsb` is a separete problem. See #560 for solution for this.

**Comment by HW42** [comment]
Thanks, commented there.

**Comment by metan-ucw** [comment]
Have you tried `make distclean`?

By convention `make clean` will not touch anything that was generated by configure see https://www.gnu.org/prep/standards/html_node/Standard-Targets.html#Standard-Targets

Well I guess that our `make distclean` will still miss some files and that should be fixed, but `make clean` works as expected here.

**Comment by HW42** [comment]
Yup, I tried distclean. I just re-checked:
```
$ make distclean
[...]
$ git clean -dxn
Would remove aclocal.m4
Would remove compile
Would remove confQgriwL/
Would remove config.guess
Would remove config.sub
Would remove configure
Would remove gitignore_clean/
Would remove include/config.h.in
Would remove include/lapi/syscalls.h
Would remove include/mk/config-openposix.mk
Would remove install-sh
Would remove m4/Makefile.in
Would remove missing
Would remove testcases/realtime/compile
Would remove testcases/realtime/config.guess
Would remove testcases/realtime/config.sub
Would remove testcases/realtime/configure
Would remove testcases/realtime/include/realtime_config.h.in
Would remove testcases/realtime/install-sh
Would remove testcases/realtime/m4/Makefile.in
Would remove testcases/realtime/missing
Would remove utils/ffsb-6.0-rc2/.deps/
Would remove utils/ffsb-6.0-rc2/Makefile
Would remove utils/ffsb-6.0-rc2/autom4te.cache/
Would remove utils/ffsb-6.0-rc2/config.h
Would remove utils/ffsb-6.0-rc2/config.h.in~
Would remove utils/ffsb-6.0-rc2/config.log
Would remove utils/ffsb-6.0-rc2/config.status
Would remove utils/ffsb-6.0-rc2/stamp-h
Would remove utils/ffsb-6.0-rc2/stamp-h1
$ 
```

**Comment by pevik** [comment]
Can we close it?

**Comment by HW42** [comment]
Sorry for not responding earlier. But this is still not fixed with current master (1dfd30c73f5d2a9af1d8ef4eb69d23ac41855344):

```
$ make distclean
[...]
$ git clean -dxn
Would remove aclocal.m4
Would remove compile
Would remove confCSjfvA/
Would remove config.guess
Would remove config.sub
Would remove configure
Would remove gitignore_clean/
Would remove include/config.h.in
Would remove include/lapi/syscalls.h
Would remove include/mk/config-openposix.mk
Would remove install-sh
Would remove m4/Makefile.in
Would remove missing
Would remove testcases/realtime/compile
Would remove testcases/realtime/config.guess
Would remove testcases/realtime/config.sub
Would remove testcases/realtime/configure
Would remove testcases/realtime/include/realtime_config.h.in
Would remove testcases/realtime/install-sh
Would remove testcases/realtime/m4/Makefile.in
Would remove testcases/realtime/missing
$ 
```

**Comment by metan-ucw** [comment]
To be honest this is low priority for the project.

But feel free to send a patch, it should be as easy as adding these files to the clean target to the respective Makefiles.
